src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -143,7 +170,30 @@ public void setStreamStatus(String streamStatus) {
 	 */
 	@Override
 	public void controlProperty(ControllableProperty controllableProperty) {
-		//TODO
+		String property = controllableProperty.getProperty();
+		String value = String.valueOf(controllableProperty.getValue());
+		Map<String, String> stats = this.localExtendedStatistics.getStatistics();
+		List<AdvancedControllableProperty> advancedControllableProperties = this.localExtendedStatistics.getControllableProperties();
+
+		if (this.logger.isDebugEnabled()) {
+			this.logger.debug("controlProperty property " + property);
+			this.logger.debug("controlProperty value " + value);
+		}
+		// Decoder control
+		String[] splitProperty = property.split(String.valueOf(DecoderConstant.HASH));
+		ControllingMetricGroup controllingGroup = ControllingMetricGroup.getByName(splitProperty[0]);
```

Comment:
please check before using splitProperty[0]

Reply:
to prevent out of bound exception

Reply:
I fixed it